presto-hive/src/main/java/com/facebook/presto/hive/pagefile/OutputStreamPageDataSink.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.pagefile;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.page.PagesSerde;
+import io.airlift.slice.OutputStreamSliceOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import static com.facebook.presto.spi.page.PagesSerdeUtil.writeSerializedPage;
+import static java.util.Objects.requireNonNull;
+
+public class OutputStreamPageDataSink
+        implements PageDataSink
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(OutputStreamPageDataSink.class).instanceSize();
+
+    private final OutputStreamSliceOutput output;
+    private final PagesSerde pagesSerde;
+
+    public OutputStreamPageDataSink(OutputStream outputStream, PagesSerde pagesSerde)
+    {
+        this.output = new OutputStreamSliceOutput(requireNonNull(outputStream, "outputStream is null"));
+        this.pagesSerde = requireNonNull(pagesSerde, "pagesSerde is null");
+    }
+
+    @Override
+    public long size()
+    {
+        return output.longSize();
+    }
+
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE + output.getRetainedSize();
+    }
+
+    @Override
+    public void write(Page dataPage)
+    {
+        writeSerializedPage(output, pagesSerde.serialize(dataPage));
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        output.flush();
+        output.close();
```

Comment:
Doesn't it flush automatically on close?

Reply:
Good catch, thank you !